src/Groups/cosets.jl

```diff
@@ -457,22 +457,42 @@ end
 Group double coset.
 Two double cosets are equal if, and only if, they contain the same elements.
 """
-struct GroupDoubleCoset{T <: GAPGroup, S <: GAPGroupElem}
+mutable struct GroupDoubleCoset{T <: GAPGroup, S <: GAPGroupElem}
```

Comment:
An alternative to making the whole struct `mutable` would be to change the field `X` to `X::Ref{GapObj}`. Then one can initialize that field to `Ref{GapObj}()` and later assign a GAp object via `my_double_coset.X[] = some_gap_obj`